rlp/sedes/serializable.py

```diff
@@ -174,39 +142,23 @@ def serialize(cls, obj):
             raise ObjectSerializationError(obj=obj, sedes=cls, list_exception=e)
 
     @classmethod
-    def deserialize(cls, serial, mutable=False, **kwargs):
+    def deserialize(cls, serial, **extra_kwargs):
         try:
             values = cls._meta.sedes.deserialize(serial)
         except ListDeserializationError as e:
             raise ObjectDeserializationError(serial=serial, sedes=cls, list_exception=e)
 
-        if mutable:
-            value_kwargs = merge_args_to_kwargs(values, {}, cls._meta.field_names)
-            return cls.create_mutable(**value_kwargs, **kwargs)
-        else:
-            immutable_args = tuple(make_immutable(arg) for arg in values)
-            value_kwargs = merge_args_to_kwargs(immutable_args, {}, cls._meta.field_names)
-            return cls.create_immutable(**value_kwargs, **kwargs)
+        args_as_kwargs = merge_args_to_kwargs(values, {}, cls._meta.field_names)
+        return cls(**args_as_kwargs, **extra_kwargs)
 
 
 def make_immutable(value):
-    if hasattr(value, 'as_immutable'):
-        return value.as_immutable()
-    elif not isinstance(value, (bytes, str)) and isinstance(value, collections.Sequence):
+    if isinstance(value, list):
```

Comment:
Can this whole method just be deleted now? Looks like `make_immutable` tests were dropped.

If not, `collections.Iterable` seems like a solid choice (it works on generators, for example, where `Sequence` and `list` would not). Or the old, ask forgiveness instead of permission:

```py
if isinstance(value, (bytes, str)):
    return value
else:
    try:
        return tuple(make_immutable(item) for item in value)
    except TypeError:
        return value

Reply:
Eh, it's not a strong opinion. Happy to leave it as is with `list`. :+1: 

The method doesn't seem like something that will get a lot of use outside pyrlp. If the need arises, we can expand accepted types later without much fuss.